README.md

```diff
@@ -1,2 +1,233 @@
+Reasonable Latency requirement outlined above is for End to End messaging.
+
+### How to deal with Network Spam?
+- One approach is to use a Friend-to-Friend (F2F) network. This is what Briar
+  and Secure Scuttlebutt (SSB) does.
+- For open DHT-based, another approach is to rely on proof of work like Whisper.
+  This isn't very practical for mobile / resource restricted devices, and
+  appears to have limited usability.
+- More approaches are likely possibly, such as traditional rate limiting, basic
+  peer reputation, payments, etc.
+- Global network attacks more relevant here than a specific node. What does this
+  imply for a DHT?
+
+### How to deal with Asynchronous Messaging?
+- One approach is to punt this problem to data sync layer.
+- Another example is Briar requiring two entities to both be online
+- Not clear that it is a necessary component of AC layer.
+- One idea is to use Aggregation Points (Xolotl, lake mixnet) as providers
+  similar to Loopix, but presumably with less HA guarantees.
+
+## Initial work
+
+###  Network
+We would like to develop a protocol that leverages the best research and
+protocols from the past in order to create something for the future of
+decentralised applications. We want a “roadmap without potholes” for providing
+stronger privacy assurances than Tor for both senders and receivers, while also
+scaling well and providing short-term message storage for offline users. In
+other words, we accept that a project this large requires a piecemeal approach
+but we shall understand and avoid design dead ends that prevent either scaling
+or rigorous analysis of anonymity properties.
+
+We think mix networks occupy a sweet spot in which nodes run extremely efficiently but rigorous analysis remains possible, if challenging.  There are designs like Tor with more efficient designs, but they cannot provide rigorous anonymity properties.  There are also numerous academic schemes designed to support rigorous analysis, but at extreme sacrifices in efficiency.  We choose a Sphinx-like packet format because it’s efficient, adaptable enough, and has good security proofs.  
+
+We leave actually analyzing our scheme for future work in collaboration with academics.  We shall however use Poison mixing and cover traffic strategy similar to Loopix, which appears analyzable although academic work to do so remains ongoing.  We concede that mix networks impose latency on users, but any faster design would definitely not admit rigorous analysis of anonymity properties, and could not credibly claim to be stronger than Tor.  
+
+We shall tweak Sphinx slightly to accommodate both receiver anonymity and
+short-term message storage simultaneously, which may require updating its
+security proofs eventually. We consider short-term message storage essential to
+user experience and advise against doing it via a second layer protocol.
+
+We foresee the public key infrastructure (PKI) being the ultimate scaling
```

Comment:
More info on PKI would be useful, as we as alternatives.

Re gossip based - would be useful to understand tradeoffs vs other approaches